plugins/modules/source_control/gitlab/gitlab_group_members.py

```diff
@@ -0,0 +1,232 @@
+    - administrator rights on the GitLab server
+options:
+    server_url:
+        description:
+            - URL of a GitLab server.
+        required: true
+        type: str
+    access_token:
+        description:
+            - A personal access token to authenticate with the GitLab API.
+        required: true
+        type: str
+    gitlab_group:
+        description:
+            - The name of the GitLab group you want to add a member to.
+        required: true
+        type: str
+    gitlab_user:
+        description:
+            - The username of the member you want to add to the GitLab group.
+        required: true
+        type: str
+    access_level:
+        description:
+            - The access level for the user.
+            - Required if I(state=present), user state is set to present.
+        type: str
+        choices: ['guest', 'reporter', 'developer', 'maintainer', 'owner']
+    state:
+        description:
+            - State of the member in the group.
+            - On C(present), it will add a user to a GitLab group.
+            - On C(absent), will remove a user from a GitLab group.
+        choices: ['present', 'absent']
+        default: 'present'
+        type: str
+notes:
+    - Supports C(check_mode).
+'''
+
+EXAMPLES = '''
+- name: Add a user to a GitLab Group
+  community.general.gitlab_group_members:
+    server_url: 'https://gitlab.example.com'
+    access_token: 'Your-Private-Token'
+    gitlab_group: groupname
+    gitlab_user: username
+    access_level: developer
+    state: present
+
+- name: Remove a user from a GitLab Group
+  community.general.gitlab_group_members:
+    server_url: 'https://gitlab.example.com'
+    access_token: 'Your-Private-Token'
+    gitlab_group: groupname
+    gitlab_user: username
+    state: absent
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import traceback
+
+try:
+    import gitlab
+    HAS_PY_GITLAB = True
+except ImportError:
+    GITLAB_IMP_ERR = traceback.format_exc()
+    HAS_PY_GITLAB = False
+
+
+class GitLabGroup(object):
+    def __init__(self, module, gl):
+        self._module = module
+        self._gitlab = gl
+
+    # check if user exists
+    def does_user_exist(self, gitlab_user):
+        return self._gitlab.users.list(username=gitlab_user)[0]
+
+    # check if group exists
+    def does_group_exist(self, gitlab_group):
+        return self._gitlab.groups.list(search=gitlab_group)[0]
+
+    # check if the user is a member of the group
+    def is_user_a_member(self, gitlab_group_id, gitlab_user_id):
+        group = self._gitlab.groups.get(gitlab_group_id)
+        members = group.members.list()
+        for member in members:
+            if member.id == gitlab_user_id:
+                return True
+        return False
+
+    # add user to a group
+    def add_member_to_group(self, gitlab_user_id, gitlab_group_id, access_level):
+        try:
+            if self._module.check_mode:
```

Comment:
Do not handle check mode in here, but in the calling code! (Same for the next method.)